test/foundry/accountBalance/accountBalance.t.sol

```diff
@@ -0,0 +1,147 @@
+        marketRegistry.addPool(address(baseToken), pool.fee());
+    }
+
+    function test_setMarketRegistry_should_emit_event() public {
+        vm.expectEmit(true, false, false, true, address(accountBalance));
+        emit MarketRegistryChanged(address(marketRegistry));
+
+        accountBalance.setMarketRegistry(address(marketRegistry));
+
+        assertEq(accountBalance.getMarketRegistry(), address(marketRegistry));
+    }
+
+    function test_getMarkPrice_should_return_index_twap_if_marketRegistry_not_set() public {
+        uint32 indexTwapInterval = clearingHouseConfig.getTwapInterval();
+        // mock index twap
+        uint256 indexTwap = 100;
+        vm.mockCall(
+            address(baseToken),
+            abi.encodeWithSelector(IIndexPrice.getIndexPrice.selector, indexTwapInterval),
+            abi.encode(indexTwap)
+        );
+        assertEq(accountBalance.getMarkPrice(address(baseToken)), indexTwap);
+    }
+
+    function test_getMarkPrice_should_return_index_twap_if_market_is_not_open() public {
+        accountBalance.setMarketRegistry(address(marketRegistry));
+
+        // mock baseToken is not open
+        vm.mockCall(address(baseToken), abi.encodeWithSelector(IBaseToken.isOpen.selector), abi.encode(false));
+
+        // mock baseToken index twap
+        uint32 indexTwapInterval = clearingHouseConfig.getTwapInterval();
+        uint256 indexTwap = 100;
+        vm.mockCall(
+            address(baseToken),
+            abi.encodeWithSelector(IIndexPrice.getIndexPrice.selector, indexTwapInterval),
+            abi.encode(indexTwap)
+        );
+
+        vm.expectCall(address(baseToken), abi.encodeWithSelector(IBaseToken.isOpen.selector));
+        assertEq(accountBalance.getMarkPrice(address(baseToken)), indexTwap);
+    }
+
+    function test_getMarkPrice_should_return_moving_average_price_if_marketRegistry_is_set() public {
+        accountBalance.setMarketRegistry(address(marketRegistry));
+
+        // mock baseToken is open
+        vm.mockCall(address(baseToken), abi.encodeWithSelector(IBaseToken.isOpen.selector), abi.encode(true));
+
+        // TODO: refactor mock code statements
+
+        // mock current market price, price = 100
+        uint256 sqrtPrice = 10;
+        uint160 sqrtPriceX96 = _toUint160(sqrtPrice.mul(FixedPoint96.Q96));
+        vm.mockCall(
+            address(pool),
+            abi.encodeWithSelector(IUniswapV3PoolState.slot0.selector),
+            abi.encode(sqrtPriceX96, 0, 0, 0, 0, 0, false)
+        );
+
+        // mock market twap(30min): price = 95
+        uint32[] memory secondsAgos = new uint32[](2);
+        secondsAgos[0] = _marketTwapInterval;
+        secondsAgos[1] = 0;
+
+        // Price: |---- 90 (10min)---|---- 95 (10min)---|---- 100 (10min)---|
+        // Tick:  |-- 45000 (10min)--|-- 45541 (10min)--|-- 46054 (10min) --|
+        int56[] memory tickCumulatives = new int56[](2);
+        tickCumulatives[0] = 0;
+        tickCumulatives[1] = 81957000;
```

Comment:
Could we just directly mockcall `getSqrtMarkTwapX96` so we don't need to test library logic which is out of scope?

Reply:
UniswapV3Broker.getSqrtMarkTwapX96 is an internal function.
vm. mockCall does not currently work on internal calls. See issue [#432](https://github.com/foundry-rs/foundry/issues/432).